src/main/java/com/icthh/xm/ms/otp/service/LoginPageRefreshableConfiguration.java

```diff
@@ -0,0 +1,60 @@
+package com.icthh.xm.ms.otp.service;
+
+import com.icthh.xm.commons.config.client.api.RefreshableConfiguration;
+import com.icthh.xm.commons.logging.aop.IgnoreLogginAspect;
+import com.icthh.xm.commons.tenant.TenantContextHolder;
+import com.icthh.xm.commons.tenant.TenantContextUtils;
+import com.icthh.xm.ms.otp.config.ApplicationProperties;
+import lombok.Getter;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang3.StringUtils;
+import org.springframework.stereotype.Component;
+import org.springframework.stereotype.Service;
+import org.springframework.util.AntPathMatcher;
+
+import java.util.Optional;
+import java.util.concurrent.ConcurrentHashMap;
+
+@Slf4j
+@RequiredArgsConstructor
+@Component
+public class LoginPageRefreshableConfiguration implements RefreshableConfiguration {
```

Comment:
Why this class in the service package? Maybe better to move it in the config?

Reply:
there is already implemented class `OtpSpecService` that in fact does similar - retrieving configuration from config service and map it to POJOs. I think `...Configuration ` is more suitable name for such classes  but i did not rename already existed class and decided to put my new one near  the old one